UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -846,28 +940,46 @@ protected virtual IEnumerator ConstructMesh(GLTF.Schema.Mesh mesh, Transform par
 				yield return ConstructMeshPrimitive(primitive, meshId, i, materialIndex);
 
 				var primitiveObj = new GameObject("Primitive");
-				var meshFilter = primitiveObj.AddComponent<MeshFilter>();
-				meshFilter.sharedMesh = _assetCache.MeshCache[meshId][i].LoadedMesh;
-				var meshRenderer = primitiveObj.AddComponent<MeshRenderer>();
 
 				MaterialCacheData materialCacheData =
 					materialIndex >= 0 ? _assetCache.MaterialCache[materialIndex] : _defaultLoadedMaterial;
 
-				UnityEngine.Material material = materialCacheData.GetContents(primitive.Attributes.ContainsKey(SemanticProperties.Color(0)));
-
-				meshRenderer.material = material;
+				UnityEngine.Material material = materialCacheData.GetContents(primitive.Attributes.ContainsKey(SemanticProperties.Color(0)));
+
+				UnityEngine.Mesh curMesh = _assetCache.MeshCache[meshId][i].LoadedMesh;
+				if (NeedsSkinnedMeshRenderer(primitive, skin))
+				{
+					var skinnedMeshRenderer = primitiveObj.AddComponent<SkinnedMeshRenderer>();
+					skinnedMeshRenderer.material = material;
+					skinnedMeshRenderer.quality = SkinQuality.Bone4;
+					/*
+					if (HasBlendShapes(primitive))
+						SetupBlendShapes(primitive);
+					*/
+					if (HasBones(skin))
+						SetupBones(skin, primitive, skinnedMeshRenderer, primitiveObj, curMesh);
```

Comment:
yield after

Reply:
better to yield to the call and yield properly in SetupBones

Reply:
Changed to coroutine as suggested